src/plugins/skills/smithing/smelting-plugin.ts

```diff
@@ -0,0 +1,269 @@
+import { Item } from '@server/world/items/item';
+import { ItemContainer } from '@server/world/items/item-container';
+import { ActionType, RunePlugin } from '@server/plugins/plugin';
+import { objectIds } from '@server/world/config/object-ids';
+import { objectAction, ObjectActionDetails } from '@server/world/actor/player/action/object-action';
+import { widgets } from '@server/world/config/widget';
+import { buttonAction, ButtonActionDetails } from '@server/world/actor/player/action/button-action';
+import { itemIds } from '@server/world/config/item-ids';
+import { Subscription } from 'rxjs';
+import { Skill } from '@server/world/actor/skills';
+import { cache } from '@server/game-server';
+import { loopingAction } from '@server/world/actor/player/action/action';
+import { animationIds } from '@server/world/config/animation-ids';
+import { soundIds } from '@server/world/config/sound-ids';
+
+export const openSmeltingInterface: objectAction = (details) => {
+    details.player.activeWidget = {
+        widgetId: widgets.furnace.widgetId,
+        type: 'CHAT',
+        closeOnWalk: true
+    };
+};
+
+interface Smeltable {
+    takesInput: boolean;
+    count: number;
+    bar: Bar;
+}
+
+export interface Bar {
+    barId: number;
+    requiredLevel: number;
+    ingredients: Item[];
+    experience: number;
+}
+
+const BRONZE : Bar = {
+    barId: itemIds.bronzeBar,
+    requiredLevel: 1,
+    experience: 6.2,
+    ingredients: [
+        { itemId: itemIds.copperOre, amount: 1 },
+        { itemId: itemIds.tinOre, amount: 1 }
+    ]
+};
+
+const BLURITE : Bar = {
+    barId: itemIds.bluriteBar,
+    requiredLevel: 8,
+    experience: 8,
+    ingredients: [
+        { itemId: itemIds.bluriteOre, amount: 1 }
+    ]
+};
+
+const IRON : Bar = {
+    barId: itemIds.ironBar,
+    requiredLevel: 15,
+    experience: 12.5,
+    ingredients: [
+        { itemId: itemIds.ironOre, amount: 1 }
+    ]
+};
+
+const SILVER : Bar = {
+    barId: itemIds.silverBar,
+    requiredLevel: 20,
+    experience: 13.6,
+    ingredients: [
+        { itemId: itemIds.silverOre, amount: 1 }
+    ]
+};
+
+const STEEL : Bar = {
+    barId: itemIds.steelBar,
+    requiredLevel: 30,
+    experience: 17.5,
+    ingredients: [
+        { itemId: itemIds.ironOre, amount: 1 },
+        { itemId: itemIds.coal, amount: 1 }
```

Comment:
I think earlier versions like this one required 2 coal.
Not positive. This was something I noticed in 2009scape as well.

Reply:
I'll look into this, thanks.